js/dark-mode-switch.js

```diff
@@ -68,9 +68,10 @@
     } else {
       delStoredPrettifyTheme()
     }
-    setTheme(getStoredPreferredTheme())
     const toggle = document.querySelector('#bd-theme')
-    toggle.checked = getStoredTheme() === 'dark'
+    const theme = getStoredPreferredTheme()
+    setTheme(theme)
```

Comment:
Hmm imho we should not save the setting unless the user actually triggers/clicks the toggle. This way, the system setting is always applied if the user does not touch the setting manually. So if the user just "wants system" then there is no need to save it and set a cookie or so.

Reply:
`setTheme` only updates the current theme, while `setStoredTheme` will persist it in the local store (no cookies - they could leak to the server). See the event handler below that, where both get called, one to apply the new theme and the other to persist the selection.